File: python/pe002.py
#!/usr/bin/env python3

""" this computes projecteuler.net problem 002 """

import sys

LIMIT = 4000000
total = 0
previous = 1
current = 1
while current < LIMIT:
    NEXT = current + previous
    if (current % 2) == 0:
        total += current
    previous = current
    current = NEXT

print(total)
sys.exit(0)

File: lua/pe002.lua
-- this computes projecteuler.net problem 002
local limit = 4000000
local total = 0
local previous = 1
local current = 1
while current < limit do
  next = current + previous
  if current % 2 == 0 then
    total = total + current 
  end
  previous = current
  current = next
end

print(total)
